### ISSUE TITLE
[BUG] unclear versioning, WASMer host throwing serialisation errors when receiving `CreateInput`

### ISSUE THREAD

**Issue by pospi** [holochain/holochain#2680]
**Describe the bug**

After upgrading to the latest HDK I seem to be hitting some errors between the host and guest:

```
11:11:30 [Tryorama - Local Conductor] info: 2023-08-20T01:11:30.430648Z ERROR holochain_wasmer_host::env: /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/holochain_wasmer_host-0.0.84/src/env.rs:94: Deserialize("invalid type: map, expected variant identifier") input_type="holochain_zome_types::entry::CreateInput"
```

**Expected behavior**

Direct use of `hdk::create` should succeed as before against a compatible core.

**System information:**

```
OS: Ubuntu 22.04

holochain (holochain 0.2.1): 3f594f1a5cef41e896b99b6b46d336d54da3299d
lair-keystore (lair_keystore 0.2.4): 43be404da0fd9d57bf4429c44def405bd6490f61
hc-launch (holochain_cli_launch 0.0.13): 75ecdd0aa191ed830cc209a984a6030e656042ff
hc-scaffold (holochain_scaffolding_cli 0.1.11): 8a63d356a0856643769adc567e078796c6509511

{"git_info":null,"cargo_pkg_version":"0.2.1","hdk_version_req":"0.2.1","hdi_version_req":"0.3.1","timestamp":"2023-08-17T04:49:25.782415550Z","hostname":"localhost","host":"x86_64-unknown-linux-gnu","target":"x86_64-unknown-linux-gnu","rustc_version":"rustc 1.66.1 (90743e729 2023-01-10)","rustflags":"","profile":"release"}

holochain_cli 0.2.1

hdi = "0.3.1"
hdk = "0.2.1"
```

**Steps to reproduce**

Updating to all the latest packages & Nix configurations.

**Additional context**

I have scaffolded a new project with `nix run github:holochain/holochain#hc-scaffold -- example forum` in an effort to reproduce in isolation. This configuration seems to be set up with Holochain 0.1.5 and these dependencies:

```
hdi = "=0.2.3"
hdk = "=0.1.3"
```

These are not the latest versions on crates.io and I am unable to get a hApp build executing successfully against Holochain 0.2.1. Please advise?

**Comment by ThetaSinner** [comment]
Hi @pospi, we're aware of this issue and have tracked down the cause at the end of last week.

It's caused by an unfortunate interaction between changes in `serde` and `rmp-serde`. Enums are being serialized as 

```
{
    "myEnum": "Variant",
}
```

but expected to be in the form

```
{
    "myEnum": {
        "type": "Variant"
    }
}
```

I am working on bumping the `serde` version that `holochain` is building against for the 0.1 and 0.2 series which will resolve the issue for people using recent versions of `serde`.

For now the workaround is to limit `serde` upgrades to the version Holochain 0.2.1 is building against. Which is [1.0.171](https://github.com/holochain/holochain/blob/develop-0.2/Cargo.lock#L6428C1-L6428C20).

So in your project's Cargo.toml you could add/update a line like `serde = ">= 1.0, <= 1.0.171`. I'll update here when we release versions that work with the latest `serde`

**Comment by pospi** [comment]
Beautiful! Thanks for the workaround, I've implemented it and it seems to be working well. Will loop back and remove it once you close this issue.

**Comment by Connoropolous** [comment]
oh yes i encountered this delightful error last week too

**Comment by ThetaSinner** [comment]
This should have been fixed now in Holochain `0.1.6` and `0.2.2`. Please upgrade to the latest version when convenient and this should problem should no longer be present

**Comment by ThetaSinner** [comment]
I believe this issue is resolved so I'm going to close it. I've opened a follow-up to try to prevent this happening again in the future
